public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx

```diff
@@ -26,44 +26,67 @@ export interface SettingsPage {
   id: string;
   title: string;
   icon: IconName;
-  render: () => React.ReactNode;
+  component: React.ReactNode;
 }
 
-export class DashboardSettings extends PureComponent<Props> {
-  onClose = () => {
-    locationService.partial({ editview: null });
-  };
-
-  onChangePage = (editview: string) => {
-    locationService.partial({ editview });
-  };
+const updateSearchParams = (init: string, partial: string) => {
```

Comment:
I think we have a utility function that does smth similar: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/location.ts#L39